File: cli/rectangle.js
#!/usr/bin/env node

const fs = require('fs');
const yaml = require('js-yaml');
const { Command } = require('commander');

function computeRectangleArea(data) {
    const a = data.a || 0;
    const b = data.b || 0;
    const c = data.c || 0;
    const d = data.d || 0;
    
    const width = Math.abs(c - a);
    const height = Math.abs(d - b);
    const area = width * height;
    
    return area;
}

function main() {
    const program = new Command();
    
    program
        .name('rectangle')
        .description('Calculate rectangle area from YAML file')
        .argument('<yaml-file>', 'Path to YAML file containing rectangle coordinates')
        .action((yamlFile) => {
            const start = performance.now();
            
            const fileContents = fs.readFileSync(yamlFile, 'utf8');
            const data = yaml.load(fileContents);
            
            const area = computeRectangleArea(data);
            
            const end = performance.now();
            
            console.log(`Rectangle area: ${area}`);
            console.log(`Time: ${(end - start).toFixed(6)} ms`);
        });
    
    program.parse();
}

main();

File: cli/rectangle.py
#!/usr/bin/env python3
import argparse
import yaml
import time

def compute_rectangle_area(data):
    a = data.get('a', 0)
    b = data.get('b', 0)
    c = data.get('c', 0)
    d = data.get('d', 0)
    
    width = abs(c - a)
    height = abs(d - b)
    area = width * height
    
    return area

def main():
    parser = argparse.ArgumentParser(description='Calculate rectangle area from YAML file')
    parser.add_argument('yaml_file', help='Path to YAML file containing rectangle coordinates')
    args = parser.parse_args()
    
    start = time.perf_counter()
    
    with open(args.yaml_file, 'r') as f:
        data = yaml.safe_load(f)
    
    area = compute_rectangle_area(data)
    
    end = time.perf_counter()
    
    print(f"Rectangle area: {area}")
    print(f"Time: {(end - start) * 1000:.6f} ms")

if __name__ == "__main__":
    main()
